### PR TITLE
Add Fortran-specific build for FFTE

### PR DESCRIPTION
**Pull request by luszczek**
I added more generic support for Fortran-specific builds that will work across Fortran compiler vendors. It worked with Open MPI Gflortran and IBM Spectrum MPI XL Fotran. This is to address #13 and there is still need to figure C interface to Fortran code.

### CODE REVIEW

**Comment by cayrols** [comment]
I am not an expert in CMake but this seems correct to me.

File: benchmarks/CMakeLists.txt [hunk 1]
```diff
@@ -50,6 +50,10 @@ macro(fiber_add_mpi_test)
     unset(_fiber_COMMAND)
 endmacro()
 
+if (FIBER_ENABLE_FFTE)
+    link_directories(${CMAKE_Fortran_IMPLICIT_LINK_DIRECTORIES})
```
Comment (cayrols) [inline benchmarks/CMakeLists.txt:54]: Just one word about this implicit variable.
What is it for?

Comment (luszczek) [reply benchmarks/CMakeLists.txt:54]: For Gfortran, `implicit` library is `-lgfortran`. For IBM, it's `-lxlf` and so on. And they are usually hidden in strange place so I need `LINK_DIRECTORIES` added to tell linker where to find them.

Comment (luszczek) [reply benchmarks/CMakeLists.txt:54]: Forgot the links: https://cmake.org/cmake/help/v3.10/variable/CMAKE_LANG_IMPLICIT_LINK_DIRECTORIES.html https://cmake.org/cmake/help/v3.10/variable/CMAKE_LANG_IMPLICIT_LINK_LIBRARIES.html

Comment (cayrols) [reply benchmarks/CMakeLists.txt:54]: Thank you for the explanation.

**Review by af-ayala — APPROVED** [review]

**Pull request merged**